pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1649,6 +1649,52 @@ func computeGrowDistribution(groups []jsonManagedInstanceGroup, newNodeCount int
 	return addCount
 }
 
+// Shrink shrinks the cluster by deleting the given VMs. This is only supported
+// for managed instance groups. Currently, nodes should only be deleted from the
+// tail of the cluster, because this function does not yet support renaming
```

Comment:
Without going through decommissioning, the ips of these deleted nodes will remain in the gossip metadata, right? Methinks we need to be cautious of these ips getting reused via new vms joining the cluster.

Reply:
True, although we've been doing this continuously on the `drt` clusters - where we set the MIGs to keep IP state. `cockroach` doesn't mind the same IP much here, as the `node` identifier itself is new for the next cockroach process that joins from this IP (the data / store is wiped, so the joining node won't conflict). The only issue with this is that the operator here will have to decommission the `cockroach` nodes or it will pollute the console. The logs do however show the fault (expecting a certain node identifier on an ip) until you decommission the node. So if you do have control, better to decommission first before shrinking.

Reply:
True, although we've been doing this continuously on the drt clusters - where we set the MIGs to keep IP state. cockroach doesn't mind the same IP much here, as the node identifier itself is new for the next cockroach process that joins from this IP (the data / store is wiped, so the joining node won't conflict). The only issue with this is that the operator here will have to decommission the cockroach nodes or it will pollute the console. The logs do however show the fault (expecting a certain node identifier on an ip) until you decommission the node. So if you do have control, better to decommission first before shrinking.

Reply:
Yes these are definitely good scenarios to consider:
For case 1: When the cluster is grown again and an IP is re-used, the store id will always increment since it's a completely new disk/store.
For case 2: The latter will work, but it's where the noise comes from. Gossip will try and connect to it as if it's one of the older node id(s) that is now technically decommissioned. Cockroach will just note it in the logs and reject the connection.
Finally, the case I guess we should be more worried about is shrinking a cluster, and not decommissioning the "crdb node id". There is the chance of an unlikely event where a completely different cluster used the same IP, and it's coincidentally at the same node id offset. Then "the shrunk" cluster could potentially make a valid connection.